include/xtensor/xcsv.hpp

```diff
@@ -198,6 +198,52 @@ namespace xt
             }
         }
     }
+
+    struct csv_config
+    {
+        char delimiter;
+        std::size_t skip_rows;
+        std::ptrdiff_t max_rows;
+        std::string comments;
+    };
+
+    class xcsv
```

Comment:
So each format would implement free `load`/`dump` functions, overloaded with their config type? But e.g. `dump_csv` doesn't need a configuration, so we should have an unused config parameter anyway just to overload the `dump` function?

Reply:
Yes the config parameter is required for obth load and dump, if not used, it will act as a tag for overloading. Notice that for the csv format, we should add facade methods instead of changing the existing ones:

```cpp
template <class E>
auto load_file(std::istream& stream, const xcsv_config& config)
{
    return load_csv(stream, config.delimiter, ...)
}

template <class E>
void dump_file(std::ostream& stream, const xexpression<E>&e, const xcsv_config&)
{
    dump_csv(stream, e);
}
```